facebook/fboss
# montblanc: config: updated PM config to support 2nd source

**Description**

Updated the platform manager config to support 2nd source based on 'VersionedPmUnitConfig' in fboss with latest source code.

Note: Currently, the sensor service is limited in supporting a second source configuration using a configuration file.

main source sensor config:
![image](https://github.com/user-attachments/assets/d9c18445-6cb0-4e24-a517-be8198ec7c23)

2nd source sensor config failed info:
![image](https://github.com/user-attachments/assets/a66c3a7f-643a-44ab-ab56-105b0f7496a3)

**Motivation**
Added 2nd source config with two interchangeable chips based on 'VersionedPmUnitConfig'.
![image](https://github.com/user-attachments/assets/b7637fe2-d113-4745-a204-4a7d7f140b0b)


**Test Plan**

- build the latest fboss bsp kmod.
- build the latest fboss platform manager binary.
- Run the platform manager with the changed config files on DVT and 2nd source DVT device normally.
- Run fw_util check unit version information.
- Run the weutil check the unit eeprom data and product sub version.

main source board tested log:
[Montblanc_replace_parts_main_source_platform_manager_log_20240822.txt](https://github.com/user-attachments/files/16809410/Montblanc_replace_parts_main_source_platform_manager_log_20240822.txt)

2nd source board tested log:
[Montblanc_replace_parts_2nd_source_platform_manager_log_20240822.txt](https://github.com/user-attachments/files/16809413/Montblanc_replace_parts_2nd_source_platform_manager_log_20240822.txt)
